support-frontend/app/utils/CheckoutValidationRules.scala

```diff
@@ -180,6 +180,17 @@ object AddressAndCurrencyValidationRules {
     ) {
       stateFromRequest.isDefined.otherwise(s"state is required for $countryFromRequest")
     } else Valid
+  def hasValidPostcodeLength(postcodeFromRequest: Option[String], addressType: String): Result = {
+    val validPostCode = postcodeFromRequest match {
+      case Some(postCode) if (postCode.length <= 20) => true
+      case None => true
+      case _ => false
```

Comment:
I wonder about simplifying this into two lines?

```scala
case Some(postCode) if (postCode.length > 20) => false
case _ => true
```

I don’t think it’s important to draw attention to the extra cases, and then you can simplify the whole thing into:

```scala
postcodeFromRequest match {
  case Some(postCode) if (postCode.length > 20) =>
    Invalid(s"$addressType PostCode length must be less than 20 characters")
  case _ => Valid
}
```

(If you think it is important to have all three cases, do say! I may have missed something)

Reply:
Have refactored the function as per the comment